src/test/java/com/commercetools/sync/commons/helpers/AssetReferenceResolverTest.java

```diff
@@ -0,0 +1,204 @@
+package com.commercetools.sync.commons.helpers;
+
+import com.commercetools.sync.commons.exceptions.ReferenceResolutionException;
+import com.commercetools.sync.products.ProductSyncOptions;
+import com.commercetools.sync.products.ProductSyncOptionsBuilder;
+import com.commercetools.sync.services.TypeService;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.models.AssetDraft;
+import io.sphere.sdk.models.AssetDraftBuilder;
+import io.sphere.sdk.models.ResourceIdentifier;
+import io.sphere.sdk.models.SphereException;
+import io.sphere.sdk.types.CustomFieldsDraft;
+import io.sphere.sdk.types.Type;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+import static com.commercetools.sync.commons.MockUtils.getMockTypeService;
+import static io.sphere.sdk.models.LocalizedString.ofEnglish;
+import static java.util.Collections.emptyList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class AssetReferenceResolverTest {
+    private TypeService typeService;
+    private ProductSyncOptions syncOptions;
+
+    /**
+     * Sets up the services and the options needed for reference resolution.
+     */
+    @Before
+    public void setup() {
+        typeService = getMockTypeService();
+        syncOptions = ProductSyncOptionsBuilder.of(mock(SphereClient.class)).build();
+    }
+
+    @Test
+    public void resolveCustomTypeReference_WithKeysAsUuidSetAndAllowed_ShouldResolveReferences() {
+        final ProductSyncOptions productSyncOptions = ProductSyncOptionsBuilder.of(mock(SphereClient.class))
+                                                                               .allowUuidKeys(true)
+                                                                               .build();
+        final CustomFieldsDraft customFieldsDraft = CustomFieldsDraft
+            .ofTypeIdAndJson(UUID.randomUUID().toString(), new HashMap<>());
+
+        final AssetDraftBuilder assetDraftBuilder = AssetDraftBuilder.of(emptyList(), ofEnglish("assetName"))
+                                                                     .custom(customFieldsDraft);
+
+        final AssetReferenceResolver assetReferenceResolver =
+            new AssetReferenceResolver(productSyncOptions, typeService);
+
+
+        final AssetDraftBuilder resolvedDraftBuilder =
+            assetReferenceResolver.resolveCustomTypeReference(assetDraftBuilder)
+                                  .toCompletableFuture().join();
+
+        assertThat(resolvedDraftBuilder.getCustom()).isNotNull();
+        assertThat(resolvedDraftBuilder.getCustom().getType().getId()).isEqualTo("typeId");
```

Comment:
Can you also check the value of custom here

Reply:
this is not the goal of this unit test. The goals of this "unit" test is to just check if the reference (custom type reference in this case) has been resolved.